p2p/protocol/handler.go

```diff
@@ -36,12 +40,39 @@ func QuaiProtocolHandler(stream network.Stream, node QuaiP2PNode) {
 		// TODO: add logic to drop the peer
 		return
 	}
+	// Create a channel for messages
+	msgChan := make(chan []byte, msgChanSize)
+	closed := false
+
+	// Start worker goroutines
+	for i := 0; i < numWorkers; i++ {
+		go func() {
+			for message := range msgChan { // This should exit when msgChan is closed
+				handleMessage(message, stream, node)
+			}
+		}()
+	}
+	go func() {
+		channelSizeCheck := time.NewTicker(5 * time.Minute)
```

Comment:
Rather than a ticker, try to put it in the channel in case A, if its full it will go to case B and you can print.

Reply:
Updated